src/screens/splash.rs

```diff
@@ -135,8 +176,34 @@ fn tick_splash_timer(time: Res<Time>, mut timer: ResMut<SplashTimer>) {
     timer.0.tick(time.delta());
 }
 
-fn check_splash_timer(timer: ResMut<SplashTimer>, mut next_screen: ResMut<NextState<Screen>>) {
-    if timer.0.just_finished() {
+fn check_splash_timer(
+    mut commands: Commands,
+    asset_server: Res<AssetServer>,
+    mut timer: ResMut<SplashTimer>,
+    mut next_screen: ResMut<NextState<Screen>>,
+    mut splash_screens: ResMut<SplashScreenImageList>,
+    containers: Query<Entity, With<SplashScreenContainer>>,
+) {
+    if !timer.0.just_finished() {
+        return;
+    }
+
+    // try to get  the next splash screen. If there isn't one, we move on to the next screen
+    if let Some(next_splash_image) = splash_screens.0.pop_front() {
```

Comment:
`let-else` then set the next screen and `return` here to remove one level of indentation.